MapboxCoreNavigationTests/RouteControllerTests.swift

```diff
@@ -113,4 +134,69 @@ class RouteControllerTests: XCTestCase {
         
         XCTAssertEqual(navigation.location!.course, mbTestHeading, "Course should be using bearing")
     }
+
+    func testUserWithinTunnelEntranceRadius() {
+        let navigation = tunnelSetup.routeController
+        
+        // Intersection with a tunnel roadClass
+        let intersectionLocation = navigation.routeProgress.route.legs[0].steps[1].intersections![1].location
+        let tunnelEntranceLocation = CLLocation(latitude: intersectionLocation.latitude, longitude: intersectionLocation.longitude)
+        
+        navigation.locationManager(navigation.locationManager, didUpdateLocations: [tunnelEntranceLocation])
+
+        let currentLegProgress = (navigation.routeProgress.currentLegProgress)!
+        let currentStep = currentLegProgress.currentStep
+        let currentIntersection = currentStep.intersections![currentLegProgress.stepIndex]
```

Comment:
This is incorrectly. We're using the stepIndex into the intersection array. Don't we already have something on routeProgress that stores the current intersection?